### ISSUE TITLE
Add support for DTLS new associations with existing parameters (RFC 6347, Section 4.2.8)

### ISSUE THREAD

**Issue by veganjay** [openssl/openssl#3283]
As far as I can tell, section [RFC6347, Section 4.2.8](https://tools.ietf.org/html/rfc6347#section-4.2.8) is not implemented.
To summarize:
1) A DTLS server is started
2) A DTLS client is started, and a successful DTLS connection is made
3) A DTLS client is killed (but the DTLS socket is not shutdown)
4) The DTLS client is re-started, using the same UDP source port, and the DTLS connection is not made.
Using Wireshark, I see the "Client Hello" message but the server does not respond with"Hello Verify Request".

I am using openssl1.1.0e (built from source) on Ubuntu 16.04 64-bit.
(There does not appear to be a way to specify "source port" using s_client, so I wrote a small python program to forward UDP ports.)

To demonstrate, the UDP forwarder, s_server and s_client are run.  The problem occurs when the s_client is killed (CTRL-C) and restarted.

Full example:
[dtls_4_2_8_test.zip](https://github.com/openssl/openssl/files/950022/dtls_4_2_8_test.zip)

Are there plans to add support for this in OpenSSL?  In the meantime, I am handling this on the application DTLS server side, by expiring inactive client sessions.

**Comment by mattcaswell** [comment]
You are correct - this is not supported at the moment. There aren't any current plans to add support although we would always consider patches. Adding this capability raises some challenges - I'm not sure what it would look like to an application from an API perspective.

**Comment by veganjay** [comment]
Thanks - that's what I thought.  This is probably a somewhat rare edge case.  Just to be clear, this will not occur if the client chooses a different source port (i.e. uses an ephemeral port) on subsequent connections.
